pyhap/accessory_driver.py

```diff
@@ -859,118 +861,99 @@ def set_characteristics(self, chars_query, client_addr):
         :type chars_query: dict
         """
         # TODO: Add support for chars that do no support notifications.
-        updates = {}
-        setter_results = {}
-        setter_responses = {}
-        had_error = False
-        had_write_response = False
-        expired = False
 
-        if HAP_REPR_PID in chars_query:
-            pid = chars_query[HAP_REPR_PID]
-            expire_time = self.prepared_writes.get(client_addr, {}).pop(pid, None)
-            if expire_time is None or time.time() > expire_time:
-                expired = True
+        queries = chars_query[HAP_REPR_CHARS]
 
-        for cq in chars_query[HAP_REPR_CHARS]:
-            aid, iid = cq[HAP_REPR_AID], cq[HAP_REPR_IID]
-            setter_results.setdefault(aid, {})
+        self._notify(queries, client_addr)
 
-            if HAP_REPR_WRITE_RESPONSE in cq:
-                setter_responses.setdefault(aid, {})
-                had_write_response = True
+        updates_by_accessories_services = defaultdict(lambda: defaultdict(lambda: {}))
+        results = defaultdict(lambda: defaultdict(lambda: {}))
+        char_to_iid = {}
 
-            if expired:
-                setter_results[aid][iid] = HAP_SERVER_STATUS.INVALID_VALUE_IN_REQUEST
-                had_error = True
-                continue
+        expired = False
+        pid = chars_query.get(HAP_REPR_PID, None)
```

Comment:
```suggestion
        pid = chars_query.get(HAP_REPR_PID)
```

`None` is already the default

Reply:
I reverted expired calculation to original format